service/src/main/java/com/commercetools/pspadapter/payone/notification/common/CaptureNotificationProcessor.java

```diff
@@ -28,37 +30,64 @@ public CaptureNotificationProcessor(TenantFactory tenantFactory, TenantConfig te
+                .build();
+    }
+
+    /**
+     * Add {@link ChangeTransactionState} action to {@code actionsBuilder} if current transaction is still in different
+     * from state in {@code notification}.
+     */
+    protected ImmutableList.Builder<UpdateAction<Payment>> updateChargeTransactionState(
+            @Nonnull final Transaction transaction,
+            @Nonnull final Notification notification,
+            @Nonnull final ImmutableList.Builder<UpdateAction<Payment>> actionsBuilder) {
+        final TransactionState newTransactionState = notification.getTransactionStatus().getCtTransactionState();
+        if (transaction.getState() != newTransactionState) {
+            actionsBuilder.add(ChangeTransactionState.of(newTransactionState, transaction.getId()));
+        }
+
+        return actionsBuilder;
+    }
+
+    /**
+     * For payments except {@code BANK_TRANSFER-ADVANCE} ({@link ClearingType#PAYONE_VOR}) add {@link AddTransaction}
+     * update action with {@link TransactionType#CHARGE} and state equal to
+     * {@link io.sphere.sdk.payments.TransactionState Notification#getTransactionStatus()#getCtTransactionState()}
+     */
+    protected ImmutableList.Builder<UpdateAction<Payment>> createChargeTransaction(
+            @Nonnull final Notification notification,
+            @Nonnull final ImmutableList.Builder<UpdateAction<Payment>> actionsBuilder,
+            @Nonnull final String sequenceNumber) {
+        final MonetaryAmount amount = MoneyImpl.of(notification.getPrice(), notification.getCurrency());
 
-                    //For all payment methods that use PayOne preauthorization and capture but only one CTP-charge
-                    //we must not create a second charge transaction
-                    //at the moment that is only "BANK_TRANSFER_ADVANCE"
-                    if (ClearingType.PAYONE_VOR != ClearingType.getClearingTypeByCode(notification.getClearingtype())) {
-                    listBuilder.add(AddTransaction.of(TransactionDraftBuilder.of(TransactionType.CHARGE, amount)
-                            .timestamp(toZonedDateTime(notification))
-                            .state(TransactionState.PENDING)
-                            .interactionId(sequenceNumber)
-                            .build()));
-                    }
+        //For all payment methods that use PayOne preauthorization and capture but only one CTP-charge
+        //we must not create a second charge transaction
+        //at the moment that is only "BANK_TRANSFER_ADVANCE"
+        if (ClearingType.PAYONE_VOR != ClearingType.getClearingTypeByCode(notification.getClearingtype())) {
+            actionsBuilder.add(AddTransaction.of(TransactionDraftBuilder.of(TransactionType.CHARGE, amount)
+                    .timestamp(toZonedDateTime(notification))
+                    .state(notification.getTransactionStatus().getCtTransactionState())
```

Comment:
Why don't you set explicitly `TransactionState.SUCCESS`?

Reply:
same as here:
https://github.com/commercetools/commercetools-payone-integration/pull/211#discussion_r165920367